ethcore/builtin/src/lib.rs

```diff
@@ -548,14 +572,39 @@ impl Bn128Pairing {
 	}
 }
 
+impl Implementation for EIP1962 {
+	/// Can fail in many cases, so leave it for an implementation
+	fn execute(&self, input: &[u8], output: &mut BytesRef) -> Result<(), &'static str> {
+		let result = eth_pairings::public_interface::API::run(&input);
+		if result.is_err() {
+			trace!("EIP1962 error: {:?}", result.err().expect("is error"));
+			return Err("Precompile call was unsuccessful");
+		}
+		let result: Vec<u8> = result.expect("some result");
+		if result.len() == 0 {
+			return Err("Precompile call returned no data");
+		}
+		if result.len() == 1 {
+			// manually made BE of Uint256(1)
+			let mut buf = [0u8; 32];
+			buf[31] = result[0];
+			output.write(0, &buf);
+		} else {
+			output.write(0, &result[..]);
```

Comment:
Is this guaranteed to be of a certain length (32 bytes?)? If so, can you do `&result[..32]` here?

Reply:
A single byte is returned iff a pairing operation is performed, so I manually put this byte into last byte of 32 bytes buffer to simulate serialization of U256. In case of other operations returned length is not equal to one, so I write directly to the output.